modules/ROOT/pages/source/payload-mode.adoc

```diff
@@ -0,0 +1,185 @@
+    "TLD": null,
+    "TLDT": null,
+    "TLT": null,
+    "TZDT": null,
+    "TOT": null,
+    "TD": null,
+    "SP": null,
+    "LB": null,
+    "LI64": null,
+    "LF64": null,
+    "LS": null,
+    "LTLD": null,
+    "LTLDT": null,
+    "LTLT": null,
+    "LZDT": null,
+    "LTOT": null,
+    "LTD": null,
+    "LSP": null
+  },
+  "timestamp": {
+    "type": "I64",
+    "B": null,
+    "I64": 1729779365447,
+    "F64": null,
+    "S": null,
+    "BA": null,
+    "TLD": null,
+    "TLDT": null,
+    "TLT": null,
+    "TZDT": null,
+    "TOT": null,
+    "TD": null,
+    "SP": null,
+    "LB": null,
+    "LI64": null,
+    "LF64": null,
+    "LS": null,
+    "LTLD": null,
+    "LTLDT": null,
+    "LTLT": null,
+    "LZDT": null,
+    "LTOT": null,
+    "LTD": null,
+    "LSP": null
+  }
+}
+
+----
+
+This mode is especially beneficial for data with complex schema requirements, as it ensures compatibility even if property types change on the Neo4j side.
+
+== Understanding the `EXTENDED` Payload Structure
+
+In `EXTENDED` mode, each property includes fields for every supported Neo4j type. Only the field corresponding to the actual property type will contain a non-null value, while all others are set to null. This structure ensures that any change in the type of a property does not cause schema enforcement errors at either the source or sink connector.
+
+[cols="1,2"]
+|===
+| Field  | Description
+
+| type   | Indicates the type of the property. Possible values include: `B`, `I64`, `F64`, `S`, `BA`, `TLD`, `TLDT`, `TLT`, `TZDT`, `TOT`, `TD`, `SP`, or their list equivalents (e.g., `LB`, `LI64`, `LF64`, `LS`, `LTLD`, etc.).
+| B      | Boolean type (true or false)
+| I64    | 64-bit integer
+| F64    | 64-bit floating point
+| S      | String
+| BA     | Byte array
+| TLD    | Temporal Local Date
+| TLDT   | Temporal Local DateTime
+| TLT    | Temporal Local Time
+| TZDT   | Temporal Zoned DateTime
+| TOT    | Temporal Offset Time
+| TD     | Temporal Duration
+| SP     | Spatial Point
+| LB, LI64, LF64, LS, LTLD, etc. | Lists of each corresponding type
+|===
+
+For example, a string field will be represented as:
+
+[source,json]
+----
+{
+  "type": "S",
+  "B": null,
+  "I64": null,
+  "F64": null,
+  "S": "actual_value",
+  ...
+}
+----
+
+== Configuration Recommendations
+
+For production environments where performance and payload simplicity are important, `COMPACT` mode is recommended. If your environment involves schema evolution, temporal data types, or other complex data requirements, `EXTENDED` mode provides the necessary structure for schema compatibility.
+
+== Compatibility with Sink Connectors
+
+The `EXTENDED` format was introduced in connector version 5.1.0 to ensure that all data published to Kafka topics adheres to a consistent schema. This prevents issues when a property changes type on the Neo4j side (e.g., a name property changes from integer to string), enabling smooth data processing across connectors and Kafka consumers.
```

Comment:
Should we also mention that if there is a Neo4j sink connector instance which is fed from a Neo4j source connector, they should stick with `EXTENDED` mode.

Reply:
sure 👍